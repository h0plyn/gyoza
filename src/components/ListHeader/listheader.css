.card-header {
  display: flex;
  justify-content: space-around;
}

.rank {
  position: relative;
  right: 0.8rem;
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 1;
}

.price {
  display: flex;
  flex: 1;
  position: relative;
  justify-content: center;
  align-items: center;
  right: 1.3rem;
}

.daily-change {
  position: relative;
  left: 1.7rem;
  flex: 1;
  cursor: pointer;
}
