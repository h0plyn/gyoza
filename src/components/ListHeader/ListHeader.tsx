import { ListHeaderProps } from '../../types';
import './listheader.css';

const ListHeader = ({ sortBy }: ListHeaderProps) => {
  return (
    <div className="card-header" data-testid='listheader'>
      <h3 className="rank" onClick={(e) => sortBy(e)}>
        Rank
      </h3>
      <h3 className="f1" onClick={(e) => sortBy(e)}>
        Coin
      </h3>
      <h3 className="price" onClick={(e) => sortBy(e)}>
        Price
      </h3>
      <h3 className="daily-change" onClick={(e) => sortBy(e)}>
        24hr
      </h3>
      <h3 className="f1" onClick={(e) => sortBy(e)}>
        Market Cap
      </h3>
    </div>
  );
};

export default ListHeader;
