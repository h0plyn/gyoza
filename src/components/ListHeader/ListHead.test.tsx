import { screen, render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ListHeader from './ListHeader';

describe('<ListHeader />', () => {
  afterEach(() => {
    cleanup();
  });

  const setup = () => {
    const utils = render(<ListHeader />);
    return { ...utils };
  };

  test('it should render the ListHeader component without crashing', () => {
    render(<ListHeader />);
  });

  test('it matches the snapshot of ListHeader', () => {
    const tree = renderer.create(<ListHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should render colum titles', () => {
    const { container } = setup();
    expect(container.textContent).toBe('RankCoinPrice24hrMarket Cap');
  });
});
