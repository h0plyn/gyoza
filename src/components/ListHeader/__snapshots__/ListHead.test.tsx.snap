// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListHeader /> it matches the snapshot of ListHeader 1`] = `
<div
  className="card-header"
  data-testid="listheader"
>
  <h3
    className="rank"
    onClick={[Function]}
  >
    Rank
  </h3>
  <h3
    className="f1"
    onClick={[Function]}
  >
    Coin
  </h3>
  <h3
    className="price"
    onClick={[Function]}
  >
    Price
  </h3>
  <h3
    className="daily-change"
    onClick={[Function]}
  >
    24hr
  </h3>
  <h3
    className="f1"
    onClick={[Function]}
  >
    Market Cap
  </h3>
</div>
`;
